### ISSUE TITLE
Question: using version 0.9.0, ParagraphGraphicFactory generates -1 values, is this by design?

### ISSUE THREAD

**Issue by amjdhsn** [FXMisc/RichTextFX#750]
ParagraphGraphicFactory started to generate -1 values at v0.9.0, which wasn't the case before.
Looking at the discussions, there was an old suggestion for such value when an error occurs, but nothing clear in the documentation.

is this by design or a bug?

Thanks.

## Environment info:

- RichTextFX Version: 0.9.0
- Operating System: macOS High Sierra
- Java version: 1.8.0_171

**Comment by JordanMartinez** [comment]
This is a workaround for an issue I discovered when implementing #687. When a `ParagraphBox` gets created, it's value starts at 0. The initial caret/selection would always be added to this "first" paragraph and then removed when that paragraph index value was changed to its actual value. As a result, the main caret and selection would not be displayed when fully rendered. Thus, I set the initial paragraph index to -1 to prevent the caret/selection being added to it until the value was correct.

So, it works this way by design, but it could be changed to workaround the issue in a different way.
